### PR TITLE
Task service and repository

### PR DESCRIPTION
**Pull request by gurug-prog**
Here I have implemented logic for task repository and service. There were some troubles with TaskDto entity and key interfaces. so I needed to add extra fields and methods. However, I have no idea what should be done with `GetTasksByDeadline()` method. Explain these moments and review my implementation please, bro.

### CODE REVIEW

File: Services/Implementations/TaskService.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	}
+
+	public IEnumerable<TaskDto> GetTasks()
+	{
+		return taskRepo.GetTasks();
+	}
+
+    public IEnumerable<TaskDto> GetTasksByDeadline()
+	{
+		throw new NotImplementedException();
```
Comment (MatthewKirik) [inline Services/Implementations/TaskService.cs]: Implement the method. It should return the TaskDtos ordered by deadline from the closest to the furthest.

File: Services/Implementations/TaskService.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	}
+
+    public IEnumerable<TaskDto> GetOverdueTasks()
+	{
+		return taskRepo.GetTasks().Where(
+			task => task.Deadline > DateTime.Now
```
Comment (MatthewKirik) [inline Services/Implementations/TaskService.cs]: It is better to move DateTime.Now to local variable, as sometimes evaluation of properties may be impossible when  converting expression tree to request.

**Review by MatthewKirik — CHANGES_REQUESTED** [review]
Please fix before merging

**Comment by gurug-prog** [comment]
Yeah, bro TU. Here are fixed changes.

**Review by MatthewKirik — APPROVED** [review]
LGTM

**Pull request merged**